### PR TITLE
Improve start/end mechanism of token auction

### PR DESCRIPTION
**Pull request by hanyunx**
As mentioned in #12.

### CODE REVIEW

**Comment by hanyunx** [comment]
Right now, `duration` increases monotonously during multiple rounds of auctions, which seems bad. Maybe we could store the original `duration` set by supervisor and set back at the start of new round?

File: contracts/NonReservedNativeTokenManager.sol [hunk 1]
```diff
@@ -54,6 +54,11 @@ contract NonReservedNativeTokenManager {
         allowMint = _allowMint;
     }
 
+    event AuctionEnded(
+        address _winner,
```
Comment (ninjaahhh) [inline contracts/NonReservedNativeTokenManager.sol]: why add the underscore for the variable name?

**Comment by ninjaahhh** [comment]
the issue with current design is that if the auction is close to an end, and a bidder who actually wants to bid the current round may encounter time delay and *accidentally* enter the next round.

we need to differentiate the bidder's intention for the case above.

File: contracts/NonReservedNativeTokenManager.sol (hunk 1 above)
Comment (hanyunx) [reply contracts/NonReservedNativeTokenManager.sol]: It's needless actually.

**Pull request merged**